### PR TITLE
Add ongoing research

### PR DESCRIPTION
**Pull request by kenchris**
***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/web-nfc/pull/535.html" title="Last updated on Apr 24, 2020, 1:37 PM UTC (0b3846a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/535/9f15d9a...kenchris:0b3846a.html" title="Last updated on Apr 24, 2020, 1:37 PM UTC (0b3846a)">Diff</a>

### CODE REVIEW

File: RESEARCH.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+
+NFC technology research
+===
+
+In order to make sure that the Web NFC API is future-proof, it is important to have a good overview of all the technologies that as part of NFC as well as those that support it. There are many technologies that are used by NFC but has broader usage and there are technologies not defined in the NFC Forum that people consider part of NFC.
+
```
Comment (zolkis) [inline RESEARCH.md]: has --> have (broader usage).
Examples for the last clause?
Split up line?

File: RESEARCH.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+This document serves as a live document for our current understanding on the NFC ecosystem.
+
+
+NFC Simple Exchange Protocol
+---
+The SNEP is a simple protocol for doing peer-to-peer (P2P) NDEF transfers. It is not supported by CoreNFC (iOS) and it is being deprecated on Android (where is branded Android Beam) due to security issues).
+
```
Comment (zolkis) [inline RESEARCH.md]: If need to quote the reason, provide link?

File: RESEARCH.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+The only way to allow some kind of peer-to-peer, like pushing NDEF from active device to active device, or even establishing a two channel communication channel, is by using HCE.
+
+### Conclusion
+
+Due to SNEP being deprecated on Android and not supported by CoreNFC, Web NFC will not be adding P2P supporting over SNEP.
```
Comment (zolkis) [inline RESEARCH.md]: What is CoreNFC? Link?

File: RESEARCH.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+Host Card Emulation
+---
+The idea of HCE was originally introduced on BlackBerry devices but today it is supported by Android out of the box.
```
Comment (zolkis) [inline RESEARCH.md]: Links would be welcome, if available.

File: RESEARCH.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+iOS supports NFC Value Added Services (VAS) [**Pending research**]
+
+MiFare support
+---
+NXP MiFare cards are quite popular and exists in different generations, with some NFC support, but also with support for proprietary protocols. As many phones have NFC readers from NXP, these additionally supports MiFare proprietary protocols. As iPhone’s NFC reader is also from NXP, CoreNFC supports MiFare protocols in addition to NFC protocols.
+
+MiFare Classic and Plus cards can be configured to support NDEF, but are not NFC compatible, meaning they do require a NFC reader chip from NXP.
+
+CoreNFC doesn’t support MiFare Classic (proprietary protocols) but people have had success reading NDEF formatted MiFare Classic cards.
+
```
Comment (zolkis) [inline RESEARCH.md]: Link to what is CoreNFC?
Maybe some more explanation would not harm, i.e. how MIFARE exposes NDEF, but it's up to you.

File: RESEARCH.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+NFC Value Added Services
+---
+iOS supports NFC Value Added Services (VAS) [**Pending research**]
+
+MiFare support
+---
+NXP MiFare cards are quite popular and exists in different generations, with some NFC support, but also with support for proprietary protocols. As many phones have NFC readers from NXP, these additionally supports MiFare proprietary protocols. As iPhone’s NFC reader is also from NXP, CoreNFC supports MiFare protocols in addition to NFC protocols.
+
+MiFare Classic and Plus cards can be configured to support NDEF, but are not NFC compatible, meaning they do require a NFC reader chip from NXP.
```
Comment (zolkis) [inline RESEARCH.md]: Not NFC _Forum_ compatible, I guess.

File: RESEARCH.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+DESFire can also be NFC T4T compatible (no NFC-B support) and thus support APDU on top of ISO-DEP on top of NFC-A.
+
+Ultralight can be NFC T2T compatible.
+
+CoreNFC separates out ISO7816 (APDU) handling as it supports it over ISO-DEP as well as proprietary MiFare protocols.
+
+### Conclusion
+
+MiFare is proprietary technology and thus is not aligned with standards based approach of the web platform, thus proprietary MiFare protocols will not be supported, but as many tags and cards sold as NFC tags are actually MiFare Classic/Plus, we have decided that implementation may support Web NFC NDEF for MiFare Classic/Plus tags/cards.
```
Comment (zolkis) [inline RESEARCH.md]: Split up line, replace second "thus" (or split sentence)?

File: RESEARCH.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+NFC Type A and B Modulation
+---
+NFC Forum T1T, T2T and T4T supports NFC-A and/or NFC-B modulation and low-level I/O communication. The T4T in addition supports ISO-DEP protocol on top of either of these, usually NFC-A and NFC-B support on T4T is optional.
+
+CoreNFC doesn’t support ways to issue raw NFC-A or NFC-B commands.
+
+### Conclusion
+
+T1T and T2T are usually used for NDEF and ISO-DEP is the most common low-level protocol for T4T so it makes sense to not support these protocols initially or at long as not supported by CoreNFC.
```
Comment (zolkis) [inline RESEARCH.md]: T1T, T2T, T4T need links or definition.

File: RESEARCH.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+Protocol support
+---
+Android supports most protocols expect SNEP which is deprecated.
```
Comment (zolkis) [inline RESEARCH.md]: except

File: RESEARCH.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+CoreNFC separates easy NDEF (only read, message directly delivered) vs harder (need to select tag, but can write). Only one reader session of any type can be active in the system at a time.
+
+The system puts additional sessions in a queue and processes them in FIFO order.
+
+### Conclusion
+
+If you are building an experience on top of NFC, then you know exactly what tech (tags, card) and protocols you support and getting notified for things you don’t support is not very useful.
+
+A discoverability API additionally makes it easy to snoop info, and it makes permission handling potentially harder - like do you need to ask permission for everything when you start scanning, or when technology is detected?
```
Comment (zolkis) [inline RESEARCH.md]: Snoop info from where? Publicly readable tags?

File: RESEARCH.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+The FIFO order can be implemented on the web but it is quite hard and presumable hard for users tounderstand, especially as all async work in one reader session should block moving on to the next reader session in the FIFO queue:
+
```
Comment (zolkis) [inline RESEARCH.md]: Split sentence. 
tounderstand --> to understand

**Comment by zolkis** [comment]
Nice work, needs more discussion.

File: RESEARCH.md (hunk 8 above)
Comment (kenchris) [reply RESEARCH.md]: Type 1 Tag :)

**Pull request closed without merging**